src/handlers/CharacterPresenceHandler.ts

```diff
@@ -0,0 +1,220 @@
+import {inject, injectable} from 'inversify';
+import {World} from '../constants/world';
+import {Zone} from '../constants/zone';
+import CharacterData from '../data/CharacterData';
+import {getLogger} from '../logger';
+import CharacterPresenceHandlerInterface from '../interfaces/CharacterPresenceHandlerInterface';
+import PopulationData from '../data/PopulationData';
+import {Faction} from '../constants/faction';
+import {jsonLogOutput} from '../utils/json';
+import MongooseModelFactory from '../factories/MongooseModelFactory';
+import {CharacterPresenceSchemaInterface} from '../models/CharacterPresenceModel';
+import {TYPES} from '../constants/types';
+import ApplicationException from '../exceptions/ApplicationException';
+import {TestCharacters} from '../constants/testCharacters';
+
+@injectable()
+export default class CharacterPresenceHandler implements CharacterPresenceHandlerInterface {
+    private static readonly logger = getLogger('CharacterPresenceHandler');
+
+    private readonly characters: Map<string, CharacterData> = new Map<string, CharacterData>();
+
+    private readonly factory: MongooseModelFactory<CharacterPresenceSchemaInterface>;
+
+    private readonly initialized = false;
+
+    private flushTimer: NodeJS.Timeout|null = null;
+
+    constructor(@inject(TYPES.characterPresenceFactory) factory: MongooseModelFactory<CharacterPresenceSchemaInterface>) {
+        this.factory = factory;
+    }
+
+    public async update(characterId: string, world: World, zone: number|null = null): Promise<boolean> {
+        // Handle Sanctuary / unrecognised zones here
+        if (zone && !Object.values(Zone).includes(zone)) {
+            CharacterPresenceHandler.logger.debug(`Discarding CharacterPresence update, unrecognized zone: ${zone}`);
+            return true;
+        }
+
+        const faction = Math.floor(Math.random() * 4) + 1; // TODO API CALL TO FIGURE THIS OUT
```

Comment:
This will require completion of Census Package faction querying. @microwavekonijn

Reply:
In which cases don't we no faction?

Reply:
This will eventually be a call to a static object of the player (retrieved from DB) which contains the faction, name etc.